Support qadd_relu on pytorch mobile

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26992 Unify quantized conv and linear tests
* **#26982 Support qadd_relu on pytorch mobile**

Summary:
Fused add+relu support
Test Plan:
python test/test_quantized.py TestQNNPackOps.test_qnnpack_add

Also,
Add torch.backends.quantized.engine = "qnnpack"
Ran
python test/test_quantized.py TestQuantizedOps.test_qadd_relu_different_qparams
python test/test_quantized.py TestQuantizedOps.test_qadd_relu_same_qparams
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D17635063](https://our.internmc.facebook.com/intern/diff/D17635063)

supriyar committed:
Support qadd_relu on pytorch mobile
Summary:
Fused add+relu support
Test Plan:
Add torch.backends.quantized.engine = "qnnpack"
Ran
python test/test_quantized.py TestQuantizedOps.test_qadd_relu_different_qparams
python test/test_quantized.py TestQuantizedOps.test_qadd_relu_same_qparams
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
No changes to the test?

supriyar committed:
Update on "Support qadd_relu on pytorch mobile"


Summary:
Fused add+relu support
Test Plan:
Add torch.backends.quantized.engine = "qnnpack"
Ran
python test/test_quantized.py TestQuantizedOps.test_qadd_relu_different_qparams
python test/test_quantized.py TestQuantizedOps.test_qadd_relu_same_qparams
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar commented:
> No changes to the test?

Forgot to add. 
Eventually we need to merge the tests, but that can be a different PR.

supriyar committed:
Update on "Support qadd_relu on pytorch mobile"


Summary:
Fused add+relu support
Test Plan:
python test/test_quantized.py TestQNNPackOps.test_qnnpack_add

Also,
Add torch.backends.quantized.engine = "qnnpack"
Ran
python test/test_quantized.py TestQuantizedOps.test_qadd_relu_different_qparams
python test/test_quantized.py TestQuantizedOps.test_qadd_relu_same_qparams
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D17635063](https://our.internmc.facebook.com/intern/diff/D17635063)

[ghstack-poisoned]

jerryzh168 reviewed (approved):
LGTM

jerryzh168 commented on test/test_quantized.py:
```diff
@@ -1858,6 +1858,13 @@ def test_qnnpack_add(self, A, zero_point, scale_A, scale_B, scale_C):
 
+            Crelu = C.copy()
+            Crelu[C < 0] = 0
+            qCrelu = _quantize(Crelu, scale_C, zero_point_C)
```
I think we should just use `torch.quantize_per_tensor` instead of using `numpy`

supriyar committed:
Update on "Support qadd_relu on pytorch mobile"


Summary:
Fused add+relu support
Test Plan:
python test/test_quantized.py TestQNNPackOps.test_qnnpack_add

Also,
Add torch.backends.quantized.engine = "qnnpack"
Ran
python test/test_quantized.py TestQuantizedOps.test_qadd_relu_different_qparams
python test/test_quantized.py TestQuantizedOps.test_qadd_relu_same_qparams
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D17635063](https://our.internmc.facebook.com/intern/diff/D17635063)

[ghstack-poisoned]

supriyar committed:
Update on "Support qadd_relu on pytorch mobile"


Summary:
Fused add+relu support
Test Plan:
python test/test_quantized.py TestQNNPackOps.test_qnnpack_add

Also,
Add torch.backends.quantized.engine = "qnnpack"
Ran
python test/test_quantized.py TestQuantizedOps.test_qadd_relu_different_qparams
python test/test_quantized.py TestQuantizedOps.test_qadd_relu_same_qparams
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D17635063](https://our.internmc.facebook.com/intern/diff/D17635063)

[ghstack-poisoned]
